### DISCUSSION TITLE
Updating a single key in the object with the help of atom

### DISCUSSION THREAD

**Discussion by bhushan-patil-official** [pmndrs/jotai#1748] [Q&A]
I have an array of objects, which is a very large object. We are mapping this array in a list in UI, the thing is every time I update any one object the complete array is updated and hence the whole component is being re-rendered. To Avoid that, what I thought is to convert the array of objects into key value pair where key will be the unique uuid consisted in the object. Based on that key I will update that specific object and hence only that component will be re-rendered which is consuming the data of that key.

Here is a small sandbox giving a rough sketch about it. Although here when I update any one key, it is re-rendering all the items mapped, I tried this in React Native. Its working fine there. I am not sure why, but on RN it is re-rendering the last time updated component as well, when you update the next component.

Please have a look at the sandbox and let me know if I have done any mistake, or is there any other efficient way to deal with this kind of problem.

https://codesandbox.io/s/distracted-leakey-ofeg5p

The idea is I don't want the parent component to re-render when a single object is updated and I don't want other components to re-render as well if any one object is updated, only that respective component should be re-rendered when updated.

Please help. Thank you.

**Comment by dai-shi** [comment]
Please have a look at https://jotai.org/docs/guides/atoms-in-atom and https://jotai.org/docs/recipes/large-objects.

**Reply by bhushan-patil-official** [reply]
Hello @dai-shi , thanks for the quick response, I tried using large object using focusAtom, although whenever I update the item, the item is updated 2 times for some reason, I checked the example given in the doc as well, I tried to create the same hook which is given in the doc, but still my component is being re-rendered for some reason. If you see I have also created a useTraceUpdate hook to trace any updates which are happening to the component which is causing the re-render, but it is also returning nothing. when I check using useEffect it shows only one render which is correct. But the consoles are printing out of useEffect.
Is there something I am doing wrong?

Here is the new sandbox:
https://codesandbox.io/s/hardcore-engelbart-235vek

**Reply by dai-shi** [reply]
If you see more console logs but one useEffect log, it's a known behavior, which I call "dobule-render-without-commits". It's an expected (designed) behavior, so no issues. If you have a heavy computation in render, it will perform bad. In such case, you should wrap heavy computation with `useMemo`.

**Reply by bhushan-patil-official** [reply]
Ohkay thanks. What do you think of the implementation for the problem statement. Any suggestions on the current implementation in the sandbox?

**Reply by dai-shi** [reply]
What's the problem again? Can you make the sandbox much smaller to explain?
